packages/kbn-es/src/utils/docker.ts

```diff
@@ -390,16 +391,30 @@ export async function maybeCreateDockerNetwork(log: ToolingLog) {
 export async function maybePullDockerImage(log: ToolingLog, image: string) {
   log.info(chalk.bold(`Checking for image: ${image}`));
 
-  await execa('docker', ['pull', image], {
-    // inherit is required to show Docker pull output
-    stdio: ['ignore', 'inherit', 'pipe'],
-  }).catch(({ message }) => {
-    const errorMessage = `Error pulling image. This is likely an issue authenticating with ${DOCKER_REGISTRY}.
+  await pRetry(
+    async () => {
+      await execa('docker', ['pull', image], {
+        // inherit is required to show Docker pull output
+        stdio: ['ignore', 'inherit', 'pipe'],
+      }).catch(({ message }) => {
+        const errorMessage = `Error pulling image. This is likely an issue authenticating with ${DOCKER_REGISTRY}.
 Visit ${chalk.bold.cyan('https://docker-auth.elastic.co/github_auth')} to login.
 
 ${message}`;
-    throw createCliError(errorMessage);
-  });
+        throw createCliError(errorMessage);
+      });
+    },
+    {
+      retries: 2,
+      onFailedAttempt: (error) => {
+        // Only retry if `connection refused` is found in the error message.
+        if (!error?.message?.includes('connection refused')) {
+          throw error;
+        }
+      },
+    }
+  );
+  // TODO: Retry here on "connection refused"
```

Comment:
is this needed still?

Reply:
good catch! Removed in [`5bd7d7c` (#189815)](https://github.com/elastic/kibana/pull/189815/commits/5bd7d7ce5d8590075d214fd82525b01fccb368e5)